katdal/__init__.py

```diff
@@ -121,7 +121,13 @@ def open(filename, ref_ant='', time_offset=0.0, **kwargs):
         Object providing :class:`DataSet` interface to file(s)
 
     """
-    filenames = [filename] if isinstance(filename, str) else filename
+    if isinstance(filename, str):
+        filenames = [filename]
+    else:
+        unexpected = set(kwargs.get('preselect', {})) - {'channels'}
```

Comment:
The `dumps` could be made to work if you used the same rechunking scheme on a dataset level (i.e. adjust the slice per dataset and maybe get rid of some component datasets entirely).

Makes me think I should simply concatenate datasets on the dask level for starters 😄  The sensors will be the only challenge left then, I hope...

Reply:
I don't see a critical use case for it right now, so even if it's possible, I don't want to hold up this PR to do it (this is blocking 5 PRs on katsdpimager that I'd like to get deployed on Wednesday).

I think there would also be a chicken-and-egg problem where you need to open the component datasets to know how many dumps they contain to know how to split up the slice across datasets before opening them.

Reply:
True!